source/views/streaming/streams/list.tsx

```diff
@@ -23,52 +23,32 @@ const styles = StyleSheet.create({
 	},
 })
 
-export const StreamListView = (): JSX.Element => {
-	let [error, setError] = React.useState<Error | null>(null)
-	let [loading, setLoading] = React.useState(true)
-	let [refreshing, setRefreshing] = React.useState(false)
-	let [streams, setStreams] = React.useState<
-		Array<{title: string; data: StreamType[]}>
-	>([])
-
-	React.useEffect(() => {
-		try {
-			getStreams().then(() => {
-				setLoading(false)
-			})
-		} catch (error) {
-			if (error instanceof Error) {
-				setError(error)
-			} else {
-				setError(new Error('unknown error - not an Error'))
-			}
-			return
-		}
-	}, [])
-
-	let refresh = async (): Promise<void> => {
-		setRefreshing(true)
-		await getStreams(true)
-		setRefreshing(false)
-	}
+const groupStreams = (entries: StreamType[]) => {
+	let grouped = groupBy(entries, (j) => j.$groupBy)
+	return toPairs(grouped).map(([title, data]) => ({title, data}))
+}
 
-	let getStreams = async (
-		reload?: boolean,
-		date: Moment = moment.tz(timezone()),
-	) => {
-		let dateFrom = date.format('YYYY-MM-DD')
-		let dateTo = date.clone().add(2, 'month').format('YYYY-MM-DD')
+const useStreams = (date: Moment = moment.tz(timezone())) => {
```

Comment:
What is the return type of this function? I feel like it should be explicit if possible. If you add that, then the call to `useStreams` inside the `StreamListView` definition could get its own type annotation so that the `data`, `error`, `reload`, `isPending`, `isInitial`, and `isLoading` fields aren't just getting pulled out of thin air.

Reply:
It'd just be `typeof useFetch<StreamType[]>`, I think. `useFetch` is doing all of the hard work here, and it's exposing the types to TS' type inference properly IIRC, so those properties aren't quite as invisible as they seem at first.

Reply:
Right, we're taking advantage of type inference. 

The types can be viewed and become less magical when looking at [react-async's useFetch](https://docs.react-async.com/api/interfaces?q=usefetc#usefetch-hook) documentation. Hovering over each of these types reflects the [type information that is within useFetch's state](https://docs.react-async.com/api/state).